src/xhydro/pmp.py

```diff
@@ -0,0 +1,609 @@
+"""Module to compute Probable Maximum Precipitation (PMP)"""
+
+import sys
+from itertools import product
+
+import dask
+import numpy as np
+import pandas as pd
+import xarray as xr
+import xclim
+from lmoments3.distr import gev
+from xclim.indices.stats import fit, parametric_quantile
+
+
+def major_precipitation_events(da, acc_day, quantil=0.9, path=None):
+    """Extracts precipitation events that exceed a given quantile
+       for a given days accumulation.
+
+    Parameters
+    ----------
+    da : xr.DataArray
+        DataArray containing the precipitation values.
+    acc_day: list
+        List of precipitation accumulation days.
+    quantil: float
+        Threshold that limits the events to those that exceed this quantile.
+    path: str, optional
+        Path where the results will be saved
+
+
+    Returns
+    -------
+    da_acc : xr.DataArray
+        DataArray containing the accumulated precipitation
+    events : xr.DataArray
+        Precipitation events that exceed the given quantile for
+        the given days accumulation.
+
+    """
```

Comment:
The numpy standard is very strict; Here's an example of what generally needs to be followed:
```suggestion
    """
    Extracts precipitation events that exceed a given quantile
       for a given days accumulation.

    Parameters
    ----------
    da : xr.DataArray
        DataArray containing the precipitation values.
    acc_day: list
        List of precipitation accumulation days.
    quantil : float
        Threshold that limits the events to those that exceed this quantile.
    path : str, optional
        Path where the results will be saved.

    Returns
    -------
    da_acc : xr.DataArray
        DataArray containing the accumulated precipitation.
    events : xr.DataArray
        Precipitation events that exceed the given quantile for
        the given days accumulation.
    """
```